### PR TITLE
Expose DelayState and declare Complex Datum as copy. 

### PR DESCRIPTION
**Pull request by hdlj**

### CODE REVIEW

File: pulse-opl/src/delay.rs [hunk 1]
```diff
@@ -94,22 +104,15 @@ pub struct Delay {
...
-        Delay { datum_type: input_fact.datum_type, buffer_shape, axis, delay, overlap }
-    }
-
     pub fn new_typed(
         input_fact: &TypedFact,
         axis: usize,
         delay: usize,
         overlap: usize,
-    ) -> TractResult<Delay> {
+    ) -> Delay {
```
Comment (kali) [inline pulse-opl/src/delay.rs:112]: mmm... changing this return type may break a few places in NNEF and pulse, i think... not a problem, this API is considered private, but we need to fix them.

**Review by kali — APPROVED** [review]
ok once nnef and pulse (and whatever else gets broken acks the API change :)

File: pulse-opl/src/delay.rs (hunk 1 above)
Comment (hdlj) [reply pulse-opl/src/delay.rs:112]: Indeed, I fixed the different places where it was used. The `new` API was returning directly `Delay` without Result. I remove this API that was a duplicate with `new_typed`.

**Comment by hdlj** [comment]
Thanks for the review !

**Pull request merged**